ctapipe/core/traits.py

```diff
@@ -423,36 +427,54 @@ def attach_subarray(self, subarray):
                         arg,
                     )
                 for tel_type in matched_tel_types:
+                    self._value_for_type[tel_type] = value
                     for tel_id in subarray.get_tel_ids_for_type(tel_type):
                         self._value_for_tel_id[tel_id] = value
             elif command == "id":
                 self._value_for_tel_id[int(arg)] = value
             else:
                 raise ValueError(f"Unrecognized command: {command}")
 
-    def __getitem__(self, tel_id: Optional[int]):
+    def __getitem__(self, tel: Optional[Union[int, str]]):
         """
         Returns the resolved parameter for the given telescope id
         """
-        if tel_id is None:
-            if self._subarray_global_value is not None:
+        if tel is None:
+            if self._subarray_global_value_set:
                 return self._subarray_global_value
-            else:
-                raise KeyError("No subarray global value set for TelescopeParameter")
+
+            raise KeyError("No subarray global value set for TelescopeParameter")
+
         if self._value_for_tel_id is None:
             raise ValueError(
                 "TelescopeParameterLookup: No subarray attached, call "
                 "`attach_subarray` first before trying to access a value by tel_id"
             )
-        try:
-            return self._value_for_tel_id[tel_id]
-        except KeyError:
-            raise KeyError(
-                f"TelescopeParameterLookup: no "
-                f"parameter value was set for telescope with tel_id="
-                f"{tel_id}. Please set it explicitly, "
-                f"or by telescope type or '*'."
-            )
+
+        if isinstance(tel, (int, np.integer)):
+            try:
+                return self._value_for_tel_id[tel]
+            except KeyError:
+                raise KeyError(
+                    f"TelescopeParameterLookup: no "
+                    f"parameter value was set for telescope with tel_id="
+                    f"{tel}. Please set it explicitly, "
+                    f"or by telescope type or '*'."
+                )
+
+        if isinstance(tel, str):
+            try:
+                if self._subarray_global_value_set:
```

Comment:
This is inconsistent with the `id` case, no?
I think it would be nicer to add the defaults for both cases in the `attach_subarray` function and avoiding the extra variable `_subarray_global_value_set`

Reply:
done